include/s3select_functions.h

```diff
@@ -1231,69 +1236,728 @@ struct _fn_in : public base_function
 struct _fn_like : public base_function
 {
 
+  bool last_char(char a)
+  {
+    if(!(a+1)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
   value res;
   std::regex compiled_regex;
 
   _fn_like(value s)
   {
-    std::string string_value = s.to_string();
-    transform(string_value);
-    compiled_regex = std::regex(string_value);
-  } 
+    std::string like_string = s.to_string();
```

Comment:
in _fn_like constructor, to use the following condition on like-expr and escape-expr
(!dynamic_cast<variable*>(expr))
&&
( dynamic_cast<variable*>(expr)->m_var_type == variable::var_t::COL_VALUE)